Mathlib/RingTheory/Artinian/Module.lean

```diff
@@ -488,30 +494,64 @@ instance isMaximal_of_isPrime (p : Ideal R) [p.IsPrime] : p.IsMaximal :=
 lemma isPrime_iff_isMaximal (p : Ideal R) : p.IsPrime ↔ p.IsMaximal :=
   ⟨fun _ ↦ isMaximal_of_isPrime p, fun h ↦ h.isPrime⟩
 
+/-- The prime spectrum is in bijection with the set of prime ideals. -/
+@[simps]
+def primeSpectrum_equiv_maximalSpectrum : PrimeSpectrum R ≃ MaximalSpectrum R where
+  toFun I := ⟨I.asIdeal, isPrime_iff_isMaximal I.asIdeal |>.mp I.isPrime⟩
+  invFun I := ⟨I.asIdeal, isPrime_iff_isMaximal I.asIdeal |>.mpr I.isMaximal⟩
+  left_inv _ := rfl
+  right_inv _ := rfl
+
+lemma primeSpectrum_equiv_maximalSpectrum_comp_asIdeal :
+    MaximalSpectrum.asIdeal ∘ primeSpectrum_equiv_maximalSpectrum.toFun =
+      PrimeSpectrum.asIdeal (R := R) := rfl
+
+lemma primeSpectrum_equiv_maximalSpectrum_inv_comp_asIdeal :
+    PrimeSpectrum.asIdeal.comp primeSpectrum_equiv_maximalSpectrum.invFun =
+      MaximalSpectrum.asIdeal (R := R) := rfl
+
+lemma primeSpectrum_asIdeal_range_eq :
+    range PrimeSpectrum.asIdeal = (range <| MaximalSpectrum.asIdeal (R := R)) :=
```

Comment:
If we have `range PrimeSpectrum.asIdeal = { I : I.IsPrime }` and the maximal ideal version, this lemma could be proved by `simp only [PrimeSpectrum.range_asIdeal, MaximalSpectrum.range_asIdeal, isPrime_iff_isMaximal]`